src/process_data.c

```diff
@@ -138,7 +138,7 @@ size_t process_retrieve_now( uint8_t **data )
     size_t rv;
 
     current = get_unix_time();
-    macs = get_current_blocked_macs();
+    macs = strdup(get_current_blocked_macs());
```

Comment:
This is incorrect as the `get_current_blocked_macs()` function already performs a `strdup()`

Reply:
For tests, it is using a mock get_current_blocked_macs() function which does not perform strdup, hence test case was crashing. Added strdup for this test function and submitted PR #155